lib/HttpClient.js

```diff
@@ -484,8 +485,10 @@ function HttpClient(options) {
             'must specify http/https protocol!'
         );
         this.url = parsedUrl;
+        this.path = parsedUrl.pathname;
```

Comment:
Since we're using `pathname` and not `path`, what happens if users call:
```
var client = restify.createJsonClient({
  url: 'http://foo.com:42/bar?foo=bar' // note the query string here
});

client.get('/foo', ...)
```
?

Is that considered invalid? If so should it throw an error? Or do we consider that it's fine to perform the request by dropping the query string?

Reply:
Good observation! Query strings are already silently thrown away (prior to this PR), with the preferred way of doing this to set a `query` option in the constructor. We don't enforce this today so while we probably should throw, I'd reserve it for a breaking change. 

In general, the idea is that the constructor options are the "default" values, and those can be overridden/stomped at verb time (`get()` or `post()`) if you pass in an option with the same name. There's some clean up we need to do to actually ensure that this is consistent for all options (as identified in this PR), but also some questions about how (or if) we want to support merging vs stomping of options - I imagine it might very well be a use case to want to always merge with default `query` instead of stomping. 

With the formalization of the URL class (https://nodejs.org/api/url.html#url_class_url) we should probably make a clean break in a new major version and jettison inconsistent and legacy options names where possible.

Reply:
Yep! It is saved to `.url`, which is then passed to raw `http|https.request()` here:
https://github.com/restify/clients/blob/master/lib/HttpClient.js#L190

However, it seems the query value which was parsed, `?foo=bar` isn't used by the http module. As far as I can tell in local testing it's thrown away. I can add a test to actually verify that behavior hasn't changed (and can revisit in said breaking change).